tests/foreman/api/test_discoveryrule.py

```diff
@@ -24,91 +24,95 @@
-            'CPU_Count = 1',
-            'disk_count < 5',
-            'memory > 500',
-            'model = KVM',
-            'Organization = Default_Organization',
-        ]
-        # Create discovery rule
-        name = gen_choice(valid_data_list())
-        search = gen_choice(searches)
-        hostname = 'myhost-<%= rand(99999) %>'
-        org = entities.Organization().create()
-        loc = entities.Location().create()
-        hostgroup = entities.HostGroup(organization=[org]).create()
-        discovery_rule = entities.DiscoveryRule(
-            name=name,
-            search_=search,
-            hostname=hostname,
-            organization=[org],
-            location=[loc],
-            hostgroup=hostgroup,
-        ).create()
-        assert name == discovery_rule.name
-        assert hostname == discovery_rule.hostname
-        assert search == discovery_rule.search_
-        assert org.name == discovery_rule.organization[0].read().name
-        assert loc.name == discovery_rule.location[0].read().name
-        assert discovery_rule.enabled is True
-        # Update discovery rule
-        name = gen_choice(valid_data_list())
-        search = 'Location = Default_Location'
-        org = entities.Organization().create()
-        loc = entities.Location().create()
-        hostgroup = entities.HostGroup(organization=[org]).create()
-        max_count = gen_integer(1, 100)
-        enabled = False
-        discovery_rule.name = name
-        discovery_rule.search_ = search
-        discovery_rule.organization = [org]
-        discovery_rule.location = [loc]
-        discovery_rule.hostgroup = hostgroup
-        discovery_rule.max_count = max_count
-        discovery_rule.enabled = enabled
-        discovery_rule = discovery_rule.update(
-            ['name', 'organization', 'location', 'hostgroup', 'search_', 'max_count', 'enabled']
-        )
-        assert name == discovery_rule.name
-        assert search == discovery_rule.search_
-        assert org.name == discovery_rule.organization[0].read().name
-        assert loc.name == discovery_rule.location[0].read().name
-        assert hostgroup.id == discovery_rule.hostgroup.id
-        assert max_count == discovery_rule.max_count
-        assert enabled == discovery_rule.enabled
-        # Delete discovery rule
-        discovery_rule.delete()
-        with pytest.raises(HTTPError):
-            discovery_rule.read()
-
-    @tier1
-    def test_negative_create_with_invalid_host_limit_and_priority(self):
-        """Create a discovery rule with invalid host limit and priority
-
-        :id: e3c7acb1-ac56-496b-ac04-2a83f66ec290
-
-        :expectedresults: Validation error should be raised
-        """
-        with pytest.raises(HTTPError):
-            entities.DiscoveryRule(max_count=gen_string('alpha')).create()
-        with pytest.raises(HTTPError):
-            entities.DiscoveryRule(priority=gen_string('alpha')).create()
+
+
+@pytest.fixture(scope="module")
```

Comment:
@dvagala , If you have module level API based fixtures, you should add them here [1] so that all the tests can use them.
[1] : https://github.com/SatelliteQE/robottelo/blob/master/robottelo/api/entity_fixtures.py